packages/fxa-shared/config.ts

```diff
@@ -0,0 +1,211 @@
+  },
+  env: {
+    default: 'production',
+    doc: 'The current node.js environment',
+    env: 'NODE_ENV',
+    format: ['development', 'test', 'stage', 'production'],
+  },
+  logging: {
+    app: { default: 'fxa-graphql-api-server' },
+    fmt: {
+      default: 'heka',
+      env: 'LOGGING_FORMAT',
+      format: ['heka', 'pretty'],
+    },
+    level: {
+      default: 'info',
+      env: 'LOG_LEVEL',
+    },
+    routes: {
+      enabled: {
+        default: true,
+        doc: 'Enable route logging. Set to false to trimming CI logs.',
+        env: 'ENABLE_ROUTE_LOGGING',
+      },
+      format: {
+        default: 'default_fxa',
+        format: ['default_fxa', 'dev_fxa', 'default', 'dev', 'short', 'tiny'],
+      },
+    },
+  },
+  image: {
+    maxSize: {
+      doc: 'Maximum bytes allow for uploads',
+      default: 1024 * 1024 * 2, // 2MB
+      env: 'IMG_UPLOADS_DEST_MAX_SIZE',
+    },
+    types: {
+      doc: 'A mapping of allowed mime types and their file signatures',
+      format: Object,
+      default: {
+        // see https://en.wikipedia.org/wiki/List_of_file_signatures
+        'image/jpeg': [
+          [0xff, 0xd8, 0xff, 0xdb],
+          [0xff, 0xd8, 0xff, 0xe0],
+          [0xff, 0xd8, 0xff, 0xe1],
+        ],
+        'image/png': [[0x89, 0x50, 0x4e, 0x47, 0x0d, 0x0a, 0x1a, 0x0a]],
+      },
+    },
+  },
+  oauth: {
+    clientId: {
+      default: '98e6508e88680e1a',
+      doc: 'OAuth client id to identify with to the profile-server',
+      env: 'OAUTH_CLIENT_ID',
+      format: String,
+    },
+  },
+  profileServer: {
+    url: {
+      doc: 'URL of fxa-aprofile-server',
+      env: 'PROFILE_SERVER_URL',
+      default: 'http://localhost:1111/v1',
+    },
+  },
+  redis: {
+    accessTokens: {
+      host: {
+        default: 'localhost',
+        env: 'ACCESS_TOKEN_REDIS_HOST',
+        format: String,
+      },
+      port: {
+        default: 6379,
+        env: 'ACCESS_TOKEN_REDIS_PORT',
+        format: 'port',
+      },
+      prefix: {
+        default: 'at:',
+        env: 'ACCESS_TOKEN_REDIS_KEY_PREFIX',
+        format: String,
+        doc: 'Key prefix for access tokens in Redis',
+      },
+    },
+  },
+  sentryDsn: {
+    default: '',
+    doc: 'Sentry DSN for error and log reporting',
+    env: 'SENTRY_DSN',
+    format: 'String',
+  },
+});
+
+// handle configuration files.  you can specify a CSV list of configuration
+// files to process, which will be overlayed in order, in the CONFIG_FILES
+// environment variable.
+
+// Need to move two dirs up as we're in the compiled directory now
+const configDir = path.dirname(path.dirname(__dirname));
+let envConfig = path.join(configDir, 'config', `${conf.get('env')}.json`);
+envConfig = `${envConfig},${process.env.CONFIG_FILES || ''}`;
+const files = envConfig.split(',').filter(fs.existsSync);
+conf.loadFile(files);
+conf.validate({ allowed: 'strict' });
+const Config = conf;
+
+export type AppConfig = typeof Config;
+export default Config;
```

Comment:
I think you moved too much in this file. I was thinking about this earlier today as well, and I think for shared components like this, what makes sense would be to have the shared component include either the config or config generator that the package using it should import and use in its local config.

For example, in this case, the docs for the `fxa-shared-db/lib/db` (a README in its dir), could say something like:

> To use the auth db in your package, import `makeMySQLConfig` from `fxa-shared-db/lib/db/config`, and put it in your packages convict setup. Then pass the config into `setupAuthDatabase`.

So only the bits needed to ease proper use would be included.